zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -634,48 +654,51 @@ fn create_proposed_transaction<DbT, ParamsT, InputsErrT, FeeRuleT, N>(
     ovk_policy: OvkPolicy,
     fee_rule: &FeeRuleT,
     min_target_height: BlockHeight,
-    prior_step_results: &[(&proposal::Step<N>, BuildResult)],
-    proposal_step: &proposal::Step<N>,
+    prior_step_results: &[(&Step<N>, BuildResult)],
+    proposal_step: &Step<N>,
+    #[cfg(feature = "transparent-inputs")] unused_transparent_outputs: &mut HashMap<
```

Comment:
Because it's necessary to remove a `StepOutput` from the map when it is consumed, in order to check that the prior outputs are not double-spent and that all ephemeral outputs have been consumed by the end. (We could do it with two append-only maps, but I don't see that that's any simpler.)

Reply:
This is mostly a personal stylistic complaint that there are now two different ways of passing in prior step results. Ordinarily with functions like this I would prefer that the only side effects are produced via the explicit `wallet_db` effect capability. Not a blocker, just a stylistic difference.

Reply:
At this point I don't want to refactor this again.

Reply:
If not changing this, then please document the argument.